### PR TITLE
Command rate limiting!

### PR DESCRIPTION
**Pull request by LW001**
Basically it saves the time when you do a command and checks whether the cooldown from limit.js in ms has passed since then.

TL;DR: limit.js is the millisecond values for each permission level for each command.

### CODE REVIEW

**Comment by Arzte** [comment]
Do this based on develop please.

File: limit.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  user: {
+    comment: 30000,
+    submit: 60000,
+    info: 100000,
+    url: 30000,
+    vote: 0,
+    help: 0,
+    stats: 0
```
Comment (Arzte) [inline limit.js]: could these not be 0? Please increase to 1s

File: limit.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  mod: {
+    comment: 20000,
+    submit: 40000,
+    info: 600000,
+    url: 60000,
+    vote: 0,
+    help: 0,
+    stats: 0
```
Comment (Arzte) [inline limit.js]: same here

**Comment by Dougley** [comment]
Some command should have a global cooldown too, so you can't use multiple accounts to spam.

**Comment by LW001** [comment]
That would so get annoying though, imagine two people wanting to do dupes
at the same time

**Comment by Dougley** [comment]
It wouldn't need to be overly huge, think a global 500ms timeout for submit.

File: index.js [hunk 3]
```diff
@@ -12,6 +12,9 @@ const bot = new Discordie({
   autoReconnect: true
 })
 
+var limits = require('./limit.js')
```
Comment (nullpixel) [inline index.js]: constant pls

**Review by nullpixel — DISMISSED** [review]

File: index.js [hunk 4]
```diff
@@ -53,7 +56,27 @@ bot.Dispatcher.on(Events.MESSAGE_CREATE, (c) => {
           if (Commands[cmd].phantom !== undefined) msg.reply('this command is restricted, and not available to you.')
           return
         } else if (level !== 2 && Commands[cmd].adminOnly === true) return
+        if (limits.global[cmd] && Date.now() - commandLimitMap.get('global') < limits.global[cmd] && level !== 2) {
```
Comment (Dougley) [inline index.js]: `commandLimitMap.get(msg.author.id+cmd)` is not guaranteed to return a value, but this is assumig it is.

**Review by Dougley — CHANGES_REQUESTED** [review]

**Pull request closed without merging**